### DISCUSSION TITLE
Have project require plugin

### DISCUSSION THREAD

**Discussion by keriksson-rosenqvist-oqc** [python-poetry/poetry#8554] [Q&A]
I have a project for which building properly depends on a plugin being installed but there is nothing preventing me running the `poetry build` command in an environment where that plugin is not installed.
I tried adding it to the build-system requires list but it had no effect.

```
[build-system]
requires = ["poetry-core>=1.0.0", "poetry-missing-plugin>=1.1.1.dev0"]
build-backend = "poetry.core.masonry.api"
```

Is there something I can add to the project's `pyproject.toml` file that would fail the build command if the plugin is not found?

**Comment by radoering** [comment]
I don't think there is something like that but I agree it would make sense. Related: #3316
